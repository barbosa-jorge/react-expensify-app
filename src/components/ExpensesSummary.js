import React from 'react';
import selectExpenses from '../selectors/expenses';
import { connect } from 'react-redux';
import numeral from 'numeral';
import getExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {

    const expensesWord = expenseCount > 1 ? 'expenses' : 'expense';
    const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00')

    return (
        <div>
            <h1>
                Viewing {expenseCount} {expensesWord} totaling {formattedExpensesTotal}
            </h1>
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters)
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: getExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);