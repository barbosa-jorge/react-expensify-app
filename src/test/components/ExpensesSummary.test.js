import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should correctly render summary expenses with no expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={0} expensesTotal={0} />)
    expect(wrapper).toMatchSnapshot()
})

test('should correctly render summary expenses with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={1000} />)
    expect(wrapper).toMatchSnapshot()
})

test('should correctly render summary expenses with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={10} expensesTotal={23226} />)
    expect(wrapper).toMatchSnapshot()
})