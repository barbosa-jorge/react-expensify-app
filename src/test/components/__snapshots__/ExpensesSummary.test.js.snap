// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render summary expenses with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    10
     
    expenses
     totaling 
    $232.26
  </h1>
</div>
`;

exports[`should correctly render summary expenses with no expenses 1`] = `
<div>
  <h1>
    Viewing 
    0
     
    expense
     totaling 
    $0.00
  </h1>
</div>
`;

exports[`should correctly render summary expenses with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totaling 
    $10.00
  </h1>
</div>
`;
